tools/src/main/java/org/apache/kafka/trogdor/common/WorkerUtils.java

```diff
@@ -70,71 +76,129 @@ public static int perSecToPerPeriod(float perSec, long periodMs) {
-                }
-                // We retry cases where the topic creation failed with a
-                // timeout.  This is a workaround for KAFKA-6368.
-                for (Map.Entry<String, Future<Void>> entry : creations.entrySet()) {
-                    String topicName = entry.getKey();
-                    Future<Void> future = entry.getValue();
-                    try {
-                        future.get();
-                        log.debug("Successfully created {}.", topicName);
-                    } catch (ExecutionException e) {
-                        if (e.getCause() instanceof TimeoutException) {
-                            log.warn("Timed out attempting to create {}: {}", topicName, e.getCause().getMessage());
-                            topicsToCreate.add(topicName);
-                        } else {
-                            log.warn("Failed to create {}", topicName, e.getCause());
-                            throw e.getCause();
+            final ScheduledExecutorService executorService = Executors.newSingleThreadScheduledExecutor(
+                ThreadUtils.createThreadFactory("createTopicsThread", false));
+            try {
+                executorService.submit(new Callable<Void>() {
+                    @Override
+                    public Void call() throws Exception {
+                        try {
+                            while (true) {
+                                final List<NewTopicCreation> batch = getBatch(MAX_CREATE_TOPICS_BATCH_SIZE);
+                                if (batch.isEmpty()) {
+                                    break;
+                                }
+                                TreeMap<String, List<String>> descriptions = new TreeMap<>();
+                                final HashMap<String, NewTopic> topics = new HashMap<>();
+                                for (NewTopicCreation newTopicCreation : batch) {
+                                    List<String> list = descriptions.get(newTopicCreation.errorMessage);
+                                    if (list == null) {
+                                        list = new ArrayList<>();
+                                        descriptions.put(newTopicCreation.errorMessage, list);
+                                    }
+                                    list.add(newTopicCreation.newTopic.name());
+                                    topics.put(newTopicCreation.newTopic.name(), newTopicCreation.newTopic);
+                                }
+                                for (Map.Entry<String, List<String>> entry : descriptions.entrySet()) {
+                                    if (!entry.getKey().isEmpty()) {
+                                        log.info("Got error {} when attempting to create topic(s): {}",
+                                            entry.getKey(), Utils.join(entry.getValue(), ", "));
+                                    }
+                                }
```

Comment:
Can we collect these errors when the createTopics responses are processed below and log them there instead? Perhaps that would simplify the code a bit?